[cleanup] fix a bunch of missing-field-initializers warnings

lrusak opened the pull request:
This PR aims to fix a bunch of missing-field-initializers warnings. This fixes all the warnings for my current build and configuration (should be a good chunk of the generic ones).

Most fixes are done by allowing the type to be default initialised using `{}`. For some reason we seem to use `{0}` a lot and I'm not sure why. This doesn't work if there are non integer types in a struct.

Some other fixes are done by modifying the default values of the struct or removing unused members all together.

I don't think there is anything wrong with we currently do but this should make sure we default initialise all members of the struct.

lrusak committed:
[cmake] enable warning -Wmissing-field-initializers

lrusak committed:
fix missing-field-initializers

yol reviewed (approved):
just looked over the wayland changes

lrusak commented:
added some more fixes including platform specific ones from the jenkins report.

I've really not sure what the best solution for `swig.cpp` is. It's a really ugly change and we don't even care about the rest of the struct (as far as I can tell). The struct also changed from python 3.8 -> 3.9. Thoughts?

garbear commented on xbmc/dialogs/GUIDialogMediaFilter.h:
```diff
@@ -44,9 +44,9 @@ class CGUIDialogMediaFilter : public CGUIDialogSettingsManualBase
-    std::shared_ptr<CSetting> setting;
-    CSmartPlaylistRule *rule;
-    void *data;
+    std::shared_ptr<CSetting> setting = nullptr;
```
This line was fine before, objects are default-constructed.

garbear replied:
I don't see what mean?

lrusak replied:
This is the error
```
/home/lukas/Documents/git/xbmc/xbmc/dialogs/GUIDialogMediaFilter.cpp:132:1: warning: missing initializer for member ‘CGUIDialogMediaFilter::Filter::setting’ [-Wmissing-field-initializers]
  132 | };
      | ^
/home/lukas/Documents/git/xbmc/xbmc/dialogs/GUIDialogMediaFilter.cpp:132:1: warning: missing initializer for member ‘CGUIDialogMediaFilter::Filter::rule’ [-Wmissing-field-initializers]
/home/lukas/Documents/git/xbmc/xbmc/dialogs/GUIDialogMediaFilter.cpp:132:1: warning: missing initializer for member ‘CGUIDialogMediaFilter::Filter::data’ [-Wmissing-field-initializers]
<snip>
```
It's because `CGUIDialogMediaFilter::Filter` doesn't get default initialized, so we either have to set initial values or set them when all the other values are set.

lrusak replied:
@garbear is this ok with you or do you have a better idea?

garbear commented on xbmc/input/touch/generic/GenericTouchActionHandler.cpp:
```diff
@@ -181,7 +181,8 @@ void CGenericTouchActionHandler::sendEvent(int actionId,
                                            int pointers /* = 1 */)
 {
-  XBMC_Event newEvent{XBMC_TOUCH};
+  XBMC_Event newEvent;
```
```c++
XBMC_Event newEvent{};
```

assuming XMBC_Event is a struct

garbear commented on xbmc/pvr/guilib/guiinfo/PVRGUIInfo.cpp:
```diff
@@ -1578,7 +1578,8 @@ void CPVRGUIInfo::CharInfoBackendNumber(std::string& strValue) const
 
   if (numBackends > 0)
-    strValue = StringUtils::Format("{0} {1} {2}", m_iCurrentActiveClient + 1, g_localizeStrings.Get(20163).c_str(), numBackends);
+    strValue = StringUtils::Format("{} {1} {2}", m_iCurrentActiveClient + 1,
```
Was this intended?

lrusak commented:
@garbear when you have a sec would you mind taking a look at the changes I've made?

thexai commented:
These have survived:

![Captura de pantalla 2021-05-24 112108](https://user-images.githubusercontent.com/58434170/119326958-c281f800-bc82-11eb-87c4-056c345e4e2f.png)

Is it intentional?

lrusak commented:
@thexai the PR aimed at fixing `missing-field-initializers` warnings not just changing `{0}` to `{}`. It was only enabled for posix platforms so if someone wants to do the equivalent on windows that would be welcome.
